src/main/java/duke/Task.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+
+import java.io.Serializable;
+
+public class Task implements Serializable {
+    private String name;
+    private boolean marked = false;
+
+    /**
+     * Returns the name of the task.
+     *
+     * @param name name of the task.
+     */
+    public Task(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Indicates the task as marked.
+     */
+    public void markTask() {
+        this.marked = true;
+    }
+
+    /**
+     * Indicates the task as un-marked.
+     */
+    public void unMarkTask() {
+        this.marked = false;
+    }
+
+    /**
+     * Returns the boolean of whether the task is marked.
+     *
+     * @return true if marked and false otherwise.
+     */
+    public boolean isMarked() {
```

Comment:
perhaps you could change this method name to something more like a verb (e.g. checkIfMarked)